src/DynamoCore/Graph/Workspaces/SerializationConverters.cs

```diff
@@ -362,6 +362,7 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
                     if(matchingNode != null)
                     {
                         matchingNode.IsSetAsInput = true;
+                        matchingNode.Name = inputData.Name;
```

Comment:
this will allow third party to access node name without the need to start a view model which is costly.